core/templates/pages/topics-and-skills-dashboard-page/create-new-skill-modal.controller.spec.ts

```diff
@@ -49,49 +49,45 @@ describe('Create new skill modal', function() {
     });
   }));
 
-  it('should init the variables', function() {
-    var rubrics = [
-      RubricObjectFactory.create(skillDifficulties[0], []),
-      RubricObjectFactory.create(skillDifficulties[1], ['']),
-      RubricObjectFactory.create(skillDifficulties[2], [])];
+  it('should initialize $scope properties after controller is initialized',
+    function() {
+      var rubrics = [
+        RubricObjectFactory.create(skillDifficulties[0], []),
+        RubricObjectFactory.create(skillDifficulties[1], ['']),
+        RubricObjectFactory.create(skillDifficulties[2], [])];
 
-    expect($scope.newSkillDescription).toEqual('');
-    expect($scope.errorMsg).toEqual('');
-    expect($scope.conceptCardExplanationEditorIsShown).toEqual(false);
-    expect($scope.bindableDict.displayedConceptCardExplanation).toEqual('');
-    expect($scope.HTML_SCHEMA).toEqual({type: 'html'});
-    expect($scope.MAX_CHARS_IN_SKILL_DESCRIPTION).toEqual(100);
-    expect($scope.newExplanationObject).toEqual(null);
-    expect($scope.rubrics).toEqual(rubrics);
-  });
+      expect($scope.newSkillDescription).toEqual('');
+      expect($scope.errorMsg).toEqual('');
+      expect($scope.conceptCardExplanationEditorIsShown).toEqual(false);
+      expect($scope.bindableDict.displayedConceptCardExplanation).toEqual('');
+      expect($scope.HTML_SCHEMA).toEqual({type: 'html'});
+      expect($scope.MAX_CHARS_IN_SKILL_DESCRIPTION).toEqual(100);
+      expect($scope.newExplanationObject).toEqual(null);
+      expect($scope.rubrics).toEqual(rubrics);
+    });
 
-  it('should open the concept card editor', function() {
+  it('should open the concept card editor when clicking on open concept card' +
+    ' explanation editor', function() {
     expect($scope.conceptCardExplanationEditorIsShown).toEqual(false);
     $scope.openConceptCardExplanationEditor();
     expect($scope.conceptCardExplanationEditorIsShown).toEqual(true);
   });
 
-  it('should reset the error message', function() {
-    expect($scope.errorMsg).toEqual('');
+  it('should update the rubrics explanation and clear error message when' +
+    ' changing skill description', function() {
     $scope.errorMsg = 'Please enter a valid description';
-    expect($scope.errorMsg).toEqual('Please enter a valid description');
-    $scope.resetErrorMsg();
+    $scope.newSkillDescription = 'Addition';
+    $scope.updateSkillDescription();
+    expect($scope.rubrics[1].getExplanations()).toEqual(['Addition']);
```

Comment:
For this and other tests that change state, make sure to add a similar pre-check before the "action". So, before updateSkillDescription(), add a pre-check that $scope.rubrics[1].getExplanations()) is equal to something else. That way, you know that the final result occurs as a result of the action (and is not just correct because of some kind of default setting).

Please fix, here and in other tests.

Reply:
Done.